src/Plugin/resource/DataProvider/DataProviderEntity.php

```diff
@@ -230,6 +230,12 @@ public function view($identifier) {
     $entity_id = $this->getEntityIdByFieldId($identifier);
 
     if (!$this->isValidEntity('view', $entity_id)) {
+      // This view may have been requested as part of a create operation. In
+      // that case, we don't throw an InaccessibleRecordException, as the create
+      // may succeed with or without the user having access to view the new item.
+      if (!$this->getResourcePath()) {
```

Comment:
We could actually check for the presence of a write method. Something like `$this->getRequest()->isWriteMethod()` (or whatever the actual method is).